mkosi/__init__.py

```diff
@@ -943,6 +943,7 @@ def build_initrd(state: MkosiState) -> Path:
         *(["--timezone", state.config.timezone] if state.config.timezone else []),
         *(["--hostname", state.config.hostname] if state.config.hostname else []),
         *(["--root-password", rootpwopt] if rootpwopt else []),
+        *([f"--environment={k}=\"{v}\"" for k, v in state.config.environment.items()]),
```

Comment:
nit
```suggestion
        *([f"--environment={k}='{v}'" for k, v in state.config.environment.items()]),
```
Not needing escaping is the one reason for single quotes ;)